### ISSUE TITLE
Pihole refresh blocklists on every reboot

### ISSUE THREAD

**Issue by TrashCanMan** [pi-hole/docker-pi-hole#352]
This is a... 

- [ ] Request for a new or modified feature
- [x] Issue trying to run the docker image
- [ ] Issue trying to build / test / develop the docker image

## Description

It was brought to my attention that pihole [shouldnt](https://www.reddit.com/r/pihole/comments/9r9gxq/dhcpdns/e8fefik/) refresh blocklists on every reboot, im not sure if this is normal on the docker version or if it is a bug, the docker works fine, blocks ads fine, dns answer correctly etc etc, its not an issue unless you restart the container often (since on every restart it takes a few mins to update the lists), no one should need to restart the container that often anyway but maybe this can create a future problem if its unatended.

## Expected Behavior

Update the blocklists on the expected day and hour (sunday morning iirc)

## Actual Behavior

Updates the blocklists on every reboot, being the same if i reboot the whole device (a rock64 in my case) or only reboot the pihole container.

## Possible Fix

No idea, im not even sure if there is anything to fix, maybe this is intended on the docker version for watever reason.

## Steps to Reproduce and debugging done

Create and start the container, add blocklists, reboot the container, this is my docker create command obcuring private data:
docker create --name pihole \
--restart=unless-stopped \
--net=host \
-p 53:53/tcp -p 53:53/udp \
-p 67:67/udp \
-p 80:80 \
-p 443:443 \
-v /afolderinmyexternalhdd/config/pihole/:/etc/pihole/ \
-v /afolderinmyexternalhdd/config/pihole/dnsmasq.d/:/etc/dnsmasq.d/ \
-e ServerIP=xxx.xxx.xxx.250 \
-e ServerIPv6=fd04:xxxx:xxxx::250 \
-e TZ=America/Argentina/Buenos_Aires \
-e WEBPASSWORD=mypassword \
--cap-add=NET_ADMIN \
pihole/pihole:v4.0_aarch64
docker start pihole
I see no volume or permissions issues at all and also i see nothing obvious on the logs, just that it updates blocklists on reboot (only the ones that have changes, as expected too)
e.g. your docker run command, pages to visit, CLI commands you ran
1.
2. 
3. 
4. 

## Debug steps I have tried

Ive checked the cron job list and there is only one updategravity job on sundays as expected.
Also i checked the date and time of the board and they are correct.
- [x] I have tried destroying my container instance, pulling the newest image version, and re-creating a new container
- [ ] I have tried running the nearly stock `docker run` example in the readme (removing any customizations I added)
- [ ] I have tried running without my volume data mounts to eliminate volumes as the cause
- [x] I have searched this repository for existing issues and pull requests that look similar 

## Context and extra information
IMO its not a problem, only annoying because it takes more time than expected to give back dns service, in fact i didnt know it was a weird behaviour until someone mentioned it on a reddit post...
I tried doing some google-fu to see if this behaviour is normal on docker-pihole but all i can find is that it should update only once a week.

## Your Environment 

* Docker Host Operating System and OS Version: Linux LibreELEC 4.4.154 #1 SMP Sun Oct 7 09:36:03 BST 2018 aarch64 GNU/Linux
* Docker Version:17.10.0-ce
* Hardware architecture: Its a rock64 SBC, arm64 (aarch64) architecture, 4gb ram, external hdd, nothing out of the ordinary

**Comment by diginc** [comment]
It may speed up boot up time if we only did this on first boot.  I'm not opposed to modifying it to be optional to run gravity on every container startup, otherwise only on cron/demand.

**Comment by lancehudson** [comment]
I agree with this, just getting started and so I have lots of gravity lists, some which now I am noticing are failing to load. Anyhow the point is it takes 5ish minutes before the container starts to serve traffic.

Maybe best of both worlds? Start the main processes and then update gravity? (or update in a non blocking way)

**Comment by AthrogatePEQ** [comment]
I've been digging through the container so that I can move my PiHole to a Docker and the spin up time is currently preventing me from moving.  I've found that in the /etc/cont-init.d/20-start.sh the following code looks like it was meant to be able to prevent gravity.sh from running at boot up but looks like maybe gravity.sh has changed since this was created.

`# Gotta go fast, no time for gravity
if [ -n "$PYTEST" ]; then
    sed -i 's/^gravity_spinup$/#gravity_spinup # DISABLED FOR PYTEST/g' "$(which gravity.sh)"
fi
gravity.sh
`

As I'm new to the game here, I think that if gravity.sh was moved inside that test condition then people decide if they wanted to update gravity or not on their own.
-Tyler
